code/addons/test/src/components/TestProviderRender.tsx

```diff
@@ -106,23 +107,49 @@ const statusMap: Record<TestStatus, ComponentProps<typeof TestStatusIcon>['statu
   pending: 'pending',
 };
 
-export const TestProviderRender: FC<
-  {
-    api: API;
-    state: TestProviderConfig & TestProviderState<Details, Config>;
-    entryId?: string;
-  } & ComponentProps<typeof Container>
-> = ({ state, api, entryId, ...props }) => {
+type TestProviderRenderProps = {
+  api: API;
+  state: TestProviderConfig & TestProviderState<Details, Config>;
+  entryId?: string;
+} & ComponentProps<typeof Container>;
+
+export const TestProviderRender: FC<TestProviderRenderProps> = ({
+  state,
+  api,
+  entryId,
+  ...props
+}) => {
   const [isEditing, setIsEditing] = useState(false);
   const theme = useTheme();
   const coverageSummary = state.details?.coverageSummary;
+  const storybookState = useStorybookState();
 
   const isA11yAddon = addons.experimental_getRegisteredAddons().includes(A11Y_ADDON_ID);
 
+  const isA11yAddonInitiallyChecked = useMemo(() => {
+    const internalIndex = storybookState.internal_index;
+    if (!internalIndex || !isA11yAddon) {
+      return false;
+    }
+
+    const allTags = Object.values(internalIndex.entries).reduce((acc, entry) => {
+      entry.tags?.forEach((tag: Tag) => {
+        acc.add(tag);
+      });
+      return acc;
+    }, new Set<Tag>());
+
+    if (allTags.has('a11y-test')) {
+      return true;
+    }
+
+    return false;
```

Comment:
Could this be simplified with `.some()` instead?


```suggestion
    return Object.values(internalIndex.entries).some((entry) => {
      return entry.tags?.includes('a11y-test');
    })
```

Reply:
Good catch!